a/modules/file/file.field.inc 
b/modules/file/file.field.inc

@@ -231,7 +231,11 @@ function file_field_insert($entity_type, $entity, $field, $instance, $langcode,
   list($id, $vid, $bundle) = entity_extract_ids($entity_type, $entity);
 
   // Add a new usage of each uploaded file.
-  foreach ($items as $item) {
+  foreach ($items as $delta => $item) {
+    if (empty($item['fid'])) {
+      unset($items[$delta]);
+      continue;
+    }
     $file = (object) $item;
     file_usage_add($file, 'file', $entity_type, $id);
   }

   
   
-  foreach ($items as $item) {
+  foreach ($items as $delta => $item) { 
+    if (empty($item['fid'])) {
+      unset($items[$delta]);
+      continue;
+    }
    if (!in_array($item['fid'], $original_fids)) {
      $file = (object) $item;
      file_usage_add($file, 'file', $entity_type, $id);
    }
  }